### PR TITLE
webhooks: Allow setting nil secret on update

### PR DESCRIPTION
**Pull request by ryanslade**
Ensure that when a secret is omitted we store `null` values in the database.

## Test plan

Tests updated

### CODE REVIEW

File: internal/database/webhooks.go [hunk 1]
```diff
@@ -247,6 +247,7 @@ func (s *webhookStore) Update(ctx context.Context, actorUID int32, newWebhook *t
 		encryptedSecret string
 		keyID           string
 	)
+
 	if newWebhook.Secret != nil {
 		encryptedSecret, keyID, err = newWebhook.Secret.Encrypt(ctx, s.key)
 		if err != nil || (encryptedSecret == "" && keyID == "") {
```
Comment (ryanslade) [inline internal/database/webhooks.go:253]: @mollylogue Is the check after `||` needed? We do it again a couple of lines down

Comment (ryanslade) [reply internal/database/webhooks.go:253]: I'm going to assume it was a copy paste mistake and update

**Comment by ryanslade** [comment]
I'm finding the encryption code very confusing.

It's odd to me that you use `NewUnencryptedSecret` to create something that *is* going to be encrypted.

**Comment by mollylogue** [comment]
> I'm finding the encryption code very confusing.
> 
> It's odd to me that you use `NewUnencryptedSecret` to create something that _is_ going to be encrypted.

This also really really confused me.

File: internal/database/webhooks.go (hunk 1 above)
Comment (mollylogue) [reply internal/database/webhooks.go]: yeah, no it's not needed.

**Review by mrnugget — APPROVED** [review]

**Comment by ryanslade** [comment]
> > I'm finding the encryption code very confusing.
> > It's odd to me that you use `NewUnencryptedSecret` to create something that _is_ going to be encrypted.
> 
> This also really really confused me.

I added some doc comments which I hope make things clearer.

**Pull request merged**